overlord/servicestate/servicestate_test.go

```diff
@@ -373,3 +374,67 @@ func (s *snapServiceOptionsSuite) TestSnapServiceOptionsQuotaGroups(c *C) {
 		QuotaGroup:   grp,
 	})
 }
+
+func (s *snapServiceOptionsSuite) TestServiceControlTaskSummaries(c *C) {
+	st := s.state
+	st.Lock()
+	defer st.Unlock()
+
+	si := snap.SideInfo{RealName: "foo", Revision: snap.R(1)}
+	snp := &snap.Info{SideInfo: si}
+	snapstate.Set(st, "foo", &snapstate.SnapState{
+		Active:   true,
+		Sequence: []*snap.SideInfo{&si},
+		Current:  snap.R(1),
+		SnapType: "app",
+	})
+	appInfos := []*snap.AppInfo{
+		{
+			Snap:        snp,
+			Name:        "svc1",
+			Daemon:      "simple",
+			DaemonScope: snap.UserDaemon,
+		},
+		{
+			Snap:        snp,
+			Name:        "svc2",
+			Daemon:      "simple",
+			DaemonScope: snap.UserDaemon,
+		},
+	}
+
+	for _, tc := range []struct {
+		instruction     *servicestate.Instruction
+		expectedSummary string
+	}{
+		{
+			&servicestate.Instruction{Action: "start"},
+			`Run service command "start" for services ["svc1" "svc2"] of snap "foo"`,
+		},
+		{
+			&servicestate.Instruction{Action: "restart"},
+			`Run service command "restart" for running services of snap "foo"`,
+		},
+		{
+			&servicestate.Instruction{
+				Action: "restart",
+				Names:  []string{"foo.svc2"},
+			},
+			`Run service command "restart" for services ["svc2"] of snap "foo"`,
```

Comment:
Thanks for this test! Could you also add a variant for restart with Reload:true flag set? This would give a slightly different summary (Run service command "reload-or-restart"...)?

Reply:
Done :-)